src/routes/safe/container/Hooks/useLoadStore.jsx

```diff
@@ -0,0 +1,31 @@
+// @flow
+import { useEffect } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+
+import loadAddressBookFromStorage from '~/logic/addressBook/store/actions/loadAddressBookFromStorage'
+import addViewedSafe from '~/logic/currentSession/store/actions/addViewedSafe'
+import fetchLatestMasterContractVersion from '~/routes/safe/store/actions/fetchLatestMasterContractVersion'
+import fetchSafe from '~/routes/safe/store/actions/fetchSafe'
+import fetchTransactions from '~/routes/safe/store/actions/fetchTransactions'
+import { safeParamAddressFromStateSelector } from '~/routes/safe/store/selectors'
+
+export const useLoadStore = (setSafeLoaded: Function) => {
+  const dispatch = useDispatch()
+  const safeAddress = useSelector(safeParamAddressFromStateSelector)
+
+  useEffect(() => {
+    const fetchData = () => {
+      if (safeAddress) {
```

Comment:
so in `useLoadStore` it fetches data only if the safeAddress is present (=user is on safe page). Same question again, why is it on the root level and not in the safe route in that case?

Reply:
also I still don't understand why safeLoaded variable is placed there too when the only route that cares if safe is loaded or not in our application is /safes/*address/ route

Reply:
I misunderstood your first question, yes you are right, now I moved it to the right place :)

Reply:
Update: doing that generates the problem that I've described here:
https://github.com/gnosis/safe-react/pull/738#issuecomment-616606836

I do recommend moving it again to the `routes/index.jsx`

Please check the redux actions every time you change the tab the `useLoadStore` reloads the store

Reply:
Yes, because currently it's a regular function with a hook inside it (= it will be called on every render). You can pass safeAddress as a dependency to it and not use a selector inside

Reply:
also you can return a hook from there probably

Reply:
so the point im trying to make is that the placement of this thing is not the problem, but how it's being used is